cognite/neat/_data_model/importers/_api_importer.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any
+
+from pydantic import ValidationError
+
+from cognite.neat._client import NeatClient
+from cognite.neat._data_model.importers._base import DMSImporter
+from cognite.neat._data_model.models.dms import (
+    DataModelReference,
+    RequestSchema,
+)
+from cognite.neat._exceptions import CDFAPIException, DataModelImportException
+from cognite.neat._issues import ModelSyntaxError
+from cognite.neat._utils.http_client import FailedRequestMessage
+from cognite.neat._utils.text import humanize_collection
+from cognite.neat._utils.validation import humanize_validation_error
+
+
+class DMSAPIImporter(DMSImporter):
+    """Imports DMS in the API format."""
+
+    def __init__(self, schema: RequestSchema | dict[str, Any]) -> None:
+        self._schema = schema
+
+    def to_data_model(self) -> RequestSchema:
+        if isinstance(self._schema, RequestSchema):
+            return self._schema
+        try:
+            return RequestSchema.model_validate(self._schema)
+        except ValidationError as e:
+            humanized_errors = humanize_validation_error(e)
+            errors = [ModelSyntaxError(message=error) for error in humanized_errors]
+            raise DataModelImportException(errors) from None
```

Comment:
These are not yet covered by tests, will do so when I add a `from_yaml` method to this importer.